tests/server/test_ldap.py

```diff
@@ -0,0 +1,97 @@
+
+from imcsdk.apis.admin.ldap import ldap_configure, ldap_settings_exist,\
+        ldap_role_group_create, ldap_role_group_exists, ldap_role_group_delete,\
+        ldap_certificate_management_enable, ldap_certificate_management_disable,\
+        is_ldap_certificate_management_enabled
+
+handle = None
+
+
+def setup_module():
+    global handle
+    handle = custom_setup()
+
+
+def teardown_module():
+    global handle
+    custom_teardown(handle)
+
+
+ldap_servers = [{"id": 1, "ip": "192.168.1.1", "port": 400},
+                {"id": 2, "ip": "192.168.1.2", "port": 500},
+                {"id": 3, "ip": "192.168.1.3", "port": 600},
+                {"id": 4, "ip": "1.2.3.4", "port": 700}]
+
+ldap_servers_2 = [{"id": 1, "ip": "192.168.1.1", "port": 400},
+                  {"id": 2, "ip": "192.168.1.2", "port": 500},
+                  {"id": 3, "ip": "192.168.1.3", "port": 600},
+                  {"id": 4, "ip": "1.2.3.4", "port": 700},
+                  {"id": 5, "ip": "192.168.1.5", "port": 600},
+                  {"id": 6, "ip": "192.168.1.63", "port": 600},
+                  {"id": 7, "ip": "192.168.1.13", "port": 600}]
+
+
+def test_ldap_configure():
+    ldap_configure(
+        handle, enabled=True,
+        basedn='DC=QATCSLABTPI02,DC=cisco,DC=com',
+        domain='QATCSLABTPI02.cisco.com',
+        timeout=20, group_auth=True,
+        bind_dn='CN=administrator,CN=Users,DC=QATCSLABTPI02,DC=cisco,DC=com',
+        password='abcdefg', ldap_servers=ldap_servers)
+    match, mo = ldap_settings_exist(
+        handle, enabled=True,
+        basedn='DC=QATCSLABTPI02,DC=cisco,DC=com',
+        domain='QATCSLABTPI02.cisco.com',
+        timeout=20, group_auth=True,
+        bind_dn='CN=administrator,CN=Users,DC=QATCSLABTPI02,DC=cisco,DC=com',
+        password='abcdefg', ldap_servers=ldap_servers)
+    assert_equal(match, True)
+
+
+@raises(Exception)
+def test_ldap_configure_invalid_servers():
+    ldap_configure(
+        handle, enabled=True,
+        basedn='DC=QATCSLABTPI02,DC=cisco,DC=com',
+        domain='QATCSLABTPI02.cisco.com',
+        timeout=20, group_auth=True,
+        bind_dn='CN=administrator,CN=Users,DC=QATCSLABTPI02,DC=cisco,DC=com',
+        password='abcdefg', ldap_servers=ldap_servers_2)
+
+
+def test_ldap_role_group_create():
+    ldap_role_group_create(handle, domain='abcd.pqrs.com', name='abcd', role='user')
+    match, mo = ldap_role_group_exists(handle, domain='abcd.pqrs.com', name='abcd', role='user')
+    assert_equal(match, True)
+
+
+def test_ldap_role_group_delete():
+    ldap_role_group_delete(handle, domain='abcd.pqrs.com', name='abcd')
+    match, mo = ldap_role_group_exists(handle, domain='abcd.pqrs.com', name='abcd', role='user')
+    assert_equal(match, False)
+
+
+def test_ldap_cert_mgmt_enable():
+    ldap_certificate_management_enable(handle)
+    assert_equal(is_ldap_certificate_management_enabled(handle), True)
+
+
+def test_ldap_cert_mgmt_disable():
```

Comment:
as a part of the tests, can you also write some error cases. input missing etc..